src/Compilers/Core/Portable/PEWriter/MetadataVisitor.cs

```diff
@@ -49,7 +49,13 @@ public void Visit(IEnumerable<ICustomAttribute> customAttributes)
         public virtual void Visit(ICustomAttribute customAttribute)
         {
             this.Visit(customAttribute.GetArguments(Context));
-            this.Visit(customAttribute.Constructor(Context));
+
+            IMethodReference constructor = customAttribute.Constructor(Context, reportDiagnostics: false);
+            if (constructor != null)
```

Comment:
>if (constructor != null) [](start = 12, length = 24)

Consider doing this check on entry and returning without doing anything if ```constructor``` is null.